### DISCUSSION TITLE
Videos included in the primary language load normally but do not load in the secondary language.

### DISCUSSION THREAD

**Discussion by sandsoncosta** [hbstack/.github#94] [Q&A] [answered]
I added a secondary language to my blog and the video doesn't load properly in the secondary language.

Principal language:
![image](https://github.com/user-attachments/assets/6e3d504d-106e-40ce-9ac9-273fda97dd34)

Secondary language:
![image](https://github.com/user-attachments/assets/058bc1d9-938c-49c7-949b-e9ba8054b60e)

http://localhost/pt-br/blog/2024/11/problemas-de-seguran%C3%A7a-no-mremoteng-que-podem-deixar-sua-empresa-vulner%C3%A1vel/
https://sandsoncosta.github.io/en/blog/2024/11/security-issues-in-mremoteng-that-can-leave-your-business-vulnerable/

**Comment by razonyang** [comment, marked as answer]
The reason is that the `poc.mp4` was published under in the same path of the default language content.

For example,

|  | Path |
| ---- | --- |
| Default Language | `/blog/foo` |
| Resource | `/blog/foo/poc.mp4` |

However according to the `permalinks` config on your `hugo.yaml`.
The secondary Language content's path is different from default language (e.g. `/blog/foo-en`), which don't have the `poc.mp4` in same path, that is it.

To solve it

1) You can duplicate the resource by setting `markup.goldmark.duplicateResourceFiles`, but **I don't recommend using this approach unless you're using multihosts**.

```sh
// hugo.yaml
markup:
  goldmark:
    duplicateResourceFiles: true
```

2) Or using shortcode (which is **always recommended for any page resources**).

```html
// layouts/shortcodes/page-video.html
{{ $url := .Get 0 }}
{{ $caption := .Get 1 }}
{{ with .Page.Resources.Get $url }}
  <figure style="text-align: center;">
    <video width="640" height="340" controls>
      <source src="{{ .Permalink }}" type="{{ .MediaType }}">
      Seu navegador não suporta a tag de vídeo.
    </video>
    <figcaption><i>$caption</i></figcaption>
  </figure>
{{ end }}
```

And then replace the `<figure>...` raw HTML with the shortcode on your content.

```
{{< page-video "poc.mp4" "PoC example" >}}
```

For more advanced usage, please checkout this [doc](https://gohugo.io/templates/shortcode/).
